connectivity/check/action.go

```diff
@@ -646,3 +653,94 @@ func (a *Action) getFlows(ctx context.Context, hubbleClient observer.ObserverCli
 
 	}
 }
+
+func (a *Action) FollowFlows(ctx context.Context, pod string) {
+	hubbleClient := a.test.ctx.HubbleClient()
+	if hubbleClient == nil {
+		return
+	}
+	a.Logf("📄 Following flows for pod %s", pod)
+	filter := []*flow.FlowFilter{{SourcePod: []string{pod}}, {DestinationPod: []string{pod}}}
+	request := &observer.GetFlowsRequest{Whitelist: filter, Follow: true}
+	b, err := hubbleClient.GetFlows(ctx, request)
+	if err != nil {
+		a.Warnf("GetFlows failed: %s", err)
+		return
+	}
+
+	for {
+		res, err := b.Recv()
+		if err != nil {
+			a.Debugf("Stopped following flows for pod %s", pod)
+			return
+		}
+		switch res.GetResponseTypes().(type) {
+		case *observer.GetFlowsResponse_Flow:
+			a.flowsWithFollowMutex.Lock()
+			a.flowsWithFollow[pod] = append(a.flowsWithFollow[pod], res)
+			a.flowsWithFollowMutex.Unlock()
+		}
+	}
+}
+
+// ValidateFlowsWithFollow retrieves the flow pods of the specified pod and validates
+// that all filters find a match. On failure, t.Fail() is called.
+// An error is returned if flow validation cannot be performed.
+func (a *Action) ValidateFlowsWithFollow(ctx context.Context, pod string, reqs []filters.FlowSetRequirement) {
+	if a.test.ctx.HubbleClient() == nil {
+		return
+	}
+	var res FlowRequirementResults
+	a.Logf("📄 Matching flows for pod %s", pod)
+	for i := 0; i < defaults.FlowMaxRetries; i++ {
```

Comment:
Im curious here, would it be possible to place the flows directly into a channel? 

At which point a "flow validator" can read off the channel, select the flows they are expecting, and confirm flow is complete in a streaming fashion? 

It feels like a pipeline can be used: 

[ grpc ] -> [ channel ] -> [ validator ] 


If we want to approach streaming validation, im not sure if copying into a map is required in the long run.

Reply:
Correct, that's the general idea. The matcher currently takes a list of flow events, so we'll keep it as it is and you can move things around later to accomodate the model you described. This PR will just change the delivery/population mechanism from polling to event-based listening.

Reply:
@ldelossa i agree, i wasn't thinking about subsequent improvements we are planning to make in the flow validation logic when i wrote the code. putting flows in a channel here would be more natural.

Reply:
Got it @ti-mo SGTM